test/integration/node-specific/db.test.js

```diff
@@ -2,36 +2,33 @@
 
 const { setupDatabase, assert: test } = require(`../shared`);
 const { expect } = require('chai');
-const { Db, MongoClient } = require('../../mongodb');
+const { MongoClient } = require('../../mongodb');
 
 describe('Db', function () {
   before(function () {
     return setupDatabase(this.configuration);
   });
 
-  it('shouldCorrectlyHandleIllegalDbNames', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded'] }
-    },
-
-    test: done => {
-      const client = { bsonOptions: {} };
-      expect(() => new Db(client, 5)).to.throw('Database name must be a string');
-      expect(() => new Db(client, '')).to.throw('Database name cannot be the empty string');
-      expect(() => new Db(client, 'te$t')).to.throw(
-        "database names cannot contain the character '$'"
-      );
-      expect(() => new Db(client, '.test', function () {})).to.throw(
-        "database names cannot contain the character '.'"
-      );
-      expect(() => new Db(client, '\\test', function () {})).to.throw(
-        "database names cannot contain the character '\\'"
-      );
-      expect(() => new Db(client, 'test test', function () {})).to.throw(
-        "database names cannot contain the character ' '"
-      );
-      done();
-    }
+  context('when given illegal db name', function () {
+    let client;
+    let db;
+
+    beforeEach(function () {
+      client = this.configuration.newClient();
+    });
+
+    afterEach(async function () {
+      db = undefined;
+      await client.close();
+    });
+
+    it('should throw error on server only', async function () {
+      db = client.db('a\x00b');
+      const error = await db.createCollection('spider').catch(error => error);
+      expect(error['name']).to.equal('MongoServerError');
```

Comment:
same chai improvements

Reply:
I'll add these changes to my next commit, as well.